wave_vision/src/utils.cpp

```diff
@@ -27,4 +27,22 @@ void projection_matrix(const Mat3 &K, const Mat3 &R, const Vec3 &t, MatX &P) {
     P = K * extrinsics;
 }
 
+void convertKeypoints(const cv::KeyPoint &keypoint, Vec2 &vec_keypoints) {
+    Vec2 v(keypoint.pt.x, keypoint.pt.y);
+
+    vec_keypoints = v;
+}
+
+void convertKeypoints(const std::vector<cv::KeyPoint> &keypoints,
+                      std::vector<Vec2> &vec_keypoints) {
+    Vec2 v;
+
+    for (const auto &k : keypoints) {
+        v(0) = k.pt.x;
+        v(1) = k.pt.y;
+
+        vec_keypoints.emplace_back(v);
```

Comment:
This is not how you use `emplace_back`, should have been

```
vec_keypoints.emplace_back(k.pt.x, k.pt.y);
```

Your  version creates an unnecessary temporary variable.